torch/distributions/bernoulli.py

```diff
@@ -50,12 +50,12 @@ def __init__(self, probs=None, logits=None, validate_args=None):
     def expand(self, batch_shape, _instance=None):
         new = self._get_checked_instance(Bernoulli, _instance)
         batch_shape = torch.Size(batch_shape)
-        if 'probs' in self.__dict__:
-            new.probs = self.probs.expand(batch_shape)
-            new._param = new.probs
-        else:
+        if 'logits' in self.__dict__:
             new.logits = self.logits.expand(batch_shape)
             new._param = new.logits
+        else:
```

Comment:
If both the parametrizations have already been computed, we should copy both of them, rather than having to recompute them again later, so I would suggest changing this to:

```python
if 'probs' in self.__dict__:
  ... # expand and set
if 'logits' in self.__dict__:
 ... # expand and set
```

That will also address your original issue since it will use the initially specified logits for any log pdf computation.